models/pull.go

```diff
@@ -498,44 +498,67 @@ const (
 )
 
 // SetMerged sets a pull request to merged and closes the corresponding issue
-func (pr *PullRequest) SetMerged() (err error) {
+func (pr *PullRequest) SetMerged() (bool, error) {
 	if pr.HasMerged {
-		return fmt.Errorf("PullRequest[%d] already merged", pr.Index)
+		return false, fmt.Errorf("PullRequest[%d] already merged", pr.Index)
 	}
 	if pr.MergedCommitID == "" || pr.MergedUnix == 0 || pr.Merger == nil {
-		return fmt.Errorf("Unable to merge PullRequest[%d], some required fields are empty", pr.Index)
+		return false, fmt.Errorf("Unable to merge PullRequest[%d], some required fields are empty", pr.Index)
 	}
 
 	pr.HasMerged = true
 
 	sess := x.NewSession()
 	defer sess.Close()
-	if err = sess.Begin(); err != nil {
-		return err
+	if err := sess.Begin(); err != nil {
+		return false, err
 	}
 
-	if err = pr.loadIssue(sess); err != nil {
-		return err
+	if _, err := sess.Where("id = ?", pr.IssueID).Cols("updated_unix").Update(&Issue{
+		ID: pr.IssueID,
+	}); err != nil {
+		return false, err
 	}
 
-	if err = pr.Issue.loadRepo(sess); err != nil {
-		return err
+	if _, err := sess.Where("id = ?", pr.ID).Cols("updated_unix").Update(pr); err != nil {
+		return false, err
 	}
-	if err = pr.Issue.Repo.getOwner(sess); err != nil {
-		return err
+
+	if err := pr.loadIssue(sess); err != nil {
```

Comment:
We should make sure `pr.Issue` is `nil` before calling this function; otherwise the cached value will be used.
```suggestion
	pr.Issue = nil
	if err := pr.loadIssue(sess); err != nil {
```